File: src/main/java/com/skocur/imagecipher/encrypters/EncrypterType.java
package com.skocur.imagecipher.encrypters;

import com.skocur.imagecipher.encrypters.exceptions.InvalidEncryptionTypeException;
import org.jetbrains.annotations.NotNull;

public enum EncrypterType {
    SINGLE_LEVEL_ENCRYPTION,
    MULTI_COLOR_ENCRYPTION,
    LOW_LEVEL_BIT_ENCRYPTION,
    RSA_ENCRYPTION;

    @NotNull
    public static EncrypterType getType(int num) {
        if (num == 1) {
            return SINGLE_LEVEL_ENCRYPTION;
        } else if (num == 2) {
            return MULTI_COLOR_ENCRYPTION;
        } else if (num == 3) {
            return LOW_LEVEL_BIT_ENCRYPTION;
        } else if (num == 4) {
            return RSA_ENCRYPTION;
        } else {
            throw new InvalidEncryptionTypeException(
                    String.format("Invalid type: %d", num)
            );
        }
    }
}

File: app/src/main/kotlin/com/imagecipher/app/encrypters/EncrypterType.kt
package com.imagecipher.app.encrypters

import com.imagecipher.app.encrypters.exceptions.InvalidEncryptionTypeException

enum class EncrypterType {
    SINGLE_LEVEL_ENCRYPTION,
    MULTI_COLOR_ENCRYPTION,
    LOW_LEVEL_BIT_ENCRYPTION,
    RSA_ENCRYPTION;

    companion object {
        fun getType(num: Int): EncrypterType {
            return when (num) {
                1 -> SINGLE_LEVEL_ENCRYPTION
                2 -> MULTI_COLOR_ENCRYPTION
                3 -> LOW_LEVEL_BIT_ENCRYPTION
                4 -> RSA_ENCRYPTION
                else -> throw InvalidEncryptionTypeException("Invalid type: $num")
            }
        }
    }
}
